### PR TITLE
Added assertEq, deprecated assertEqual

### PR DESCRIPTION
**Pull request by bitemyapp**
For #1255

### CODE REVIEW

File: yesod-test/Yesod/Test.hs [hunk 1]
```diff
@@ -314,9 +319,20 @@ htmlQuery' getter errTrace query = withResponse' getter ("Tried to invoke htmlQu
...
 
--- | Asserts that the two given values are equal.
+assertEq :: (Eq a, Show a) => String -> a -> a -> YesodExample site ()
+assertEq m a b =
+  liftIO $ HUnit.assertBool msg (a == b)
+  where msg = "Assertion: " ++ m ++ "\n" ++
+              "First argument:  " ++ ppShow a ++ "\n" ++
+              "Second argument: " ++ ppShow b ++ "\n"
+
+{-# DEPRECATED assertEqual "Use assertEqual instead" #-}
```
Comment (snoyberg) [inline yesod-test/Yesod/Test.hs]: Did you mean `Use assertEq instead`?

Comment (bitemyapp) [inline yesod-test/Yesod/Test.hs]: @snoyberg quite right! Forgot to amend it after I figured out the name.

File: yesod-test/yesod-test.cabal [hunk 2]
```diff
@@ -14,29 +14,30 @@ description:     API docs and the README are available at <http://www.stackage.o
 extra-source-files: README.md, LICENSE, test/main.hs, ChangeLog.md
 
 library
-    build-depends:   base                      >= 4.3      && < 5
```
Comment (snoyberg) [inline yesod-test/yesod-test.cabal]: Were the edits to this file intentional?

Comment (bitemyapp) [inline yesod-test/yesod-test.cabal]: @snoyberg yes, my eyes swam checking for something, so I fired my re-ordering function on it. I can remove it if you like.

Comment (snoyberg) [inline yesod-test/yesod-test.cabal]: Generally I keep such changes separate, but not a big deal. I'll merge once Travis goes green.

**Pull request merged**